src/referral_maintenance.rs

```diff
@@ -22,25 +36,149 @@ pub async fn maintenaince_tasks(db: Arc<ReferralStore>, dune_data_folder: String
         {
             let mut guard = match db.0.lock() {
```

Comment:
One way to make this slightly more succinct is to use:
```rust
use anyhow::Context as _;

let mut guard = db.0.lock().context("Mutex poisoned")?;
```

Reply:
Ahhhh, I see the issue. Its because `PoisonError` can't be wrapped by `anyhow`. In that case you can:
```rust
let mut guard = db.0.lock().map_err(|_| anyhow!("Mutex poisoned"))?;
```